tedge/src/command.rs

```diff
@@ -157,10 +157,17 @@ pub trait Command {
 pub trait BuildCommand {
     fn build_command(
         self,
-        config: config::TEdgeConfig,
+        context: BuildCommandContext,
     ) -> Result<Box<dyn Command>, config::ConfigError>;
 }
 
+/// The context for `BuildCommand`
+///
+pub struct BuildCommandContext {
+    pub config: config::TEdgeConfig,
+    pub config_repository: tedge_config::TEdgeConfigRepository,
```

Comment:
Why a `config` plus a `config_repository` ?

* Is this to raise any reading error asap, i.e. before `build_command()` is called?

Reply:
I tried to avoid reading the config twice, as we unconditionally read the config. If you pass in only the `config_repository`, you have to read it again in order to store it.

I think the real issue is that we unconditionally read the `config`, before calling `BuildCommand`.

I dropped the `config` field now, because that required to `clone` the `config` in order to make it mutable.